pwndbg/search.py

```diff
@@ -17,15 +17,28 @@
 import pwndbg.typeinfo
 import pwndbg.vmmap
 
-
-def search(searchfor, mapping=None, start=None, end=None, 
+def search(searchfor, mappings=None, start=None, end=None, 
            executable=False, writable=False):
+    """Search inferior memory for a byte sequence.
+
+    Arguments:
+        searchfor(bytes): Byte sequence to find
+        mappings(list): List of pwndbg.memory.Page objects to search
+            By default, uses all available mappings.
+        start(int): First address to search, inclusive.
+        end(int): Last address to search, exclusive.
+        executable(bool): Restrict search to executable pages
+        writable(bool): Restrict search to writable pages
+
+    Yields:
+        An iterator on the address matches
+    """
     value = searchfor
     size  = None
 
     i = gdb.selected_inferior()
 
-    maps = pwndbg.vmmap.get()
+    maps = mappings or pwndbg.vmmap.get()
     hits = []
 
     if end and start:
```

Comment:
Probably would be better to change below `maps = [...]` into `maps = (...)` or maybe moving the ifs into `for`.

However both changes are not really super needed. In worst case scenario we have ~400 items in the list (tried on `google-chrome`).